src/main/java/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -33,4 +33,13 @@ public List<Book> findAll() {
             throw new EntityNotFoundException("Can't get books from DB", e);
         }
     }
+
+    @Override
+    public Book findById(Long id) {
```

Comment:
1. you should use `Optional`, since `Session` may return null
2. you can leverage `SessionFactory` method called `fromSession`

```
return Optional.ofNullable(sessionFactory.fromSession(s -> s.find(Book.class, id)));
```